test/pylib/scylla_cluster.py

```diff
@@ -531,8 +550,87 @@ def after_test(self, name) -> None:
         if self._get_keyspace_count() != self.keyspace_count:
             raise RuntimeError("Test post-condition failed, "
                                "the test must drop all keyspaces it creates.")
-        for server in self.cluster:
-            server.write_log_marker("------ Ending test {} ------\n".format(name))
+        for server in itertools.chain(self.started.values(), self.stopped.values()):
+            server.write_log_marker(f"------ Ending test {name} ------\n")
+
+    def update_last_seed(self, removed_host: str) -> None:
+        """Update last seed when removing a host"""
+        if self.last_seed == removed_host:
+            self.last_seed = next(iter(self.started.keys())) if self.started else None
+
+    async def node_stop(self, server_id: str, gracefully: bool) -> bool:
+        """Stop a server. No-op if already stopped."""
+        self.is_dirty = True
+        logging.info("Cluster %s stopping server %s", self, server_id)
+        if server_id in self.stopped or server_id in self.removed:
+            return True
+        if server_id not in self.started:
+            return False
```

Comment:
no, why, the test can do random things, including this one. we can inform the test that it's trying to stop a stopped instance over rest and it will decide what to do with it.

Reply:
Improved the logic and the result is now [success:bool, msg:str]